### DISCUSSION TITLE
Redesigning Arthropods (among other things)

### DISCUSSION THREAD

**Discussion by Venera3** [CleverRaven/Cataclysm-DDA#47275] [Ideas]
Migrated over from Issues ( #45030 ) to prevent it from being buried by the actual issues. I've updated it with the current progress, and will try to keep it up to date going forward.

The game has a wide range of insects and assorted creepy crawlies which are universally pretty boring. This comes from a few places, ranging from possible oversights ( all arthropods are blinder than a flashbanged bat ) to not utilizing later additions ( most of their special attacks are hardcoded, nothing uses the spell system) and generally bland flavor.

I'm planning on redoing all members of the spider/insect json and any other neglected monsters I feel particularly strong about and/or figure out something cool for. ~~This will all be JSON~~ The competent parts will be JSON, but I won't hold back from flailing around for low-hanging fixes.
Generally I'm aiming to give each species/group a lifecycle with at least one reproductive form to make the world feel a bit more alive and to make pacifying large swathes of land less trivial. If it makes sense I'll try to include predation/interspecies hostility, but aggroing is an all-or-nothing deal, so we'll have to see how that shakes out in game.

Lorewise the bugs aren't really hashed out, the current iteration is them being kind of an "area denial" disruptor for the Blob, which pumped them full of mutagenic juice in the days leading up to the game start and getting let loose to mostly follow their usual instincts, sowing extra chaos and denying the wilderness (where the initial zombies aren't too formadible) to survivors. They aren't directly controlled by the Blob, allowing eusocial insects to defend their nests from zombies, but the lone predatory insects got enough imprinting/bad experiences to leave zeds alone who return the curtesy. In game terms this means the previous hostility of bees, ants and wasps is now canonized, the rest got a reciprocal neutral relationship towards zombies.

### General Concepts:

- [x] Dragonflies: the naiads (larvae) are aquatic hunters with a reach attack, the adults are fast predators in medium-sized swarms, dangerous up until mid-late game with high armor piercing and a grab attack. Mega variant lays eggs, hits like a truck but can be outrun easily. - #44779

- [x] Centipedes: small, frightened young that grow slowly into the most dangerous predator of the swamps with a deadly venom and high enough damage output to threaten endgame chars. The adult's life cycle consists of flipping between the normal adult variant (threatening alone to early-midgame chars, in group giving late-midgame toons a run for their money) and the reproductive matriarchs, the blind, slow mothers giving birth to small centipedes again. At each "switch" the adults have a low chance to turn into the updated mega variant, the armored centipede that shrugs off a .223 and can bite your head off clean through power armor. - #45868

- [x] Fireflies: small, incidental critters lighting up the swamps at night, mostly useful to get chased around by the rest of the gang. Might get revisited if/when monster defenses get moved over to the spell infrastructure to get a defensive flashbang. - #45214

- [ ] Mosquitos: large swarms of hard to hit, low-damage annoyances to keep earlygame characters entertained and to give the predators something to hunt. Mega becomes a blood-gorged spiky ball that spawns smaller mosquitos (+maybe a ranged attack?). Wainting on the effects being hooked up to vitamins on them, so that they can suck your blood for real.

- [ ] Bees: add queens and drones, queens lay eggs that have a small chance to spawn a drone and a miniscule chance for a new queen. Workers ~~grow larger with age, but~~ are relatively docile until you mess with them, they also drop wax. Drones have good vision, aggro readily and have a pheromone sting to "mark" you ( if scents work, will need some testing). Queens are mean in a fight, but the main difficulty should come from numbers and the venom. 

- [ ] Wasps: egglaying queens, add a small chance for worker wasps to drop balls of flesh (randomly clean/mutated/tainted) to simulate them hunting.  Agressive from the get-go, camping out near a wasp nest should be suicidal for any non-lategame character. Both the bees and the wasps get a special "forager" class, which is less aggressive, stays neutral to zombies and will flee at low health and replaces the current "random wilderness" spawns. The spawns around their nests, however are going to be pretty territorial, zombie-ambivalent and unbreakable.

- [ ] Dermatiks: Figure out a way to implement their effect on monsters, add a zombie variant that got colonized by them but still shuffles on, maybe include them as brood parasites in the other nestbuilder's hives.

- [ ] Flies: are pretty boring, so they also become mostly prey animals. They might become the most Blob-friendly of the bunch, acting as biomass carriers / a mobile focus of blobl nastiness (maybe a low-level heal for zombies), but I'm not set on that yet

- [ ] Spiders: Divided into two groups: The solitary hunters ( jumping, wolf and cellar spiders) stay mostly the same, just get some venom updates and a new round of combat balance to keep them interesting. Cellar spiders might settle in the spacious subways. The web-spinners (web spider, trapdoor spider and black widows) become communal hunters, making their lairs in forests and houses, tended to by big scary brood mothers. Their life cycle is still TBD, probably a common spiderling diversifies into the adult morphs, with each having a chance to evolve into something scary / another brood mother. They might also become territorial enough to hurt zombies.

- [ ] Ants: combat balance, potentially redoing their life cycle to move away from the magic instahatching of eggs.

- [ ]  Expand underground wildlife: settle slugs in sewers and some new spooky centipede variants in the subways, to keep lab hunting fresh

- [ ] Other cool stuff just waiting to be included: Antlions snatching you on the shore, Millipedes eating forests and log houses, hibernating critters spawning in caves/basements in the winter, bombardier beetles with flamethrowers for butts.... you get the idea.

### Maps

- [x] Dermatik nests in the wilderness, complete with some hapless victims.

- [x] A new Radio tower variant with a wasp nest

- [ ] Redo beehives, with multiple variants, maybe one with some triffids to show that some friendships transcend dimensions

- [ ] Take a look at the anthill code to see if there's a way to move some of it over to JSON

- [ ] A nice underground/underwater nest for razorclaws to raise their roes in peace

### Misc related stuff

- [ ] Redo player venoms to actually matter

- [ ] Expand chitincrafting a bit, with chitinous plates dropping only from the bigger, scarier insects being required for proper chitin armor

- [ ] Rebalance the insect/spider mutagen recipes to use the appropriate eggs to incentivize hunting them. Maybe roll the basic mutagen into a common "arthropod" mutage and only have the serums specialize

### Describe alternatives you've considered

Accept the fact that Cataclysm is a zombie survival game. Never.

Other ideas, general feedback on the above or already merged stuff, cool critter facts and volunteers ~~for the spider pit~~ welcome!

**Comment by actual-nh** [comment]
I've been looking through various potentially-interesting Arthropods, and have just realized that I may need to go back to some I'd skipped - I skipped them for not being found as far north as Massachusetts IRL. However, ones that are currently found as far north as Virginia could well start spreading out further north as they grow larger - surface-to-volume ratio in regard to metabolic heat loss.

**Reply by Venera3** [reply]
I mean, there's already alligators so I wouldn't worry too much. Me going off on a wild tangent has been mostly kept in check by there being enough to do with the critters already in the game, but I'm 100% going to include a bombardier beetle at some point.

**Reply by actual-nh** [reply]
Point. Spitting spiders (Scytodidae) would also be nice. (And to balance things out, spiders that are known to hunt black widows...)
